app/models/manageiq/providers/redfish/physical_infra_manager/event_parser.rb

```diff
@@ -0,0 +1,21 @@
+module ManageIQ::Providers::Redfish
+  class PhysicalInfraManager::EventParser
+    def self.events_to_hashes(events, ems_id)
+      JSON.parse(events.data).fetch("Events", []).collect do |event|
```

Comment:
Hm so each event that you're adding is a json blob with data?  I think this should be expanded and queue each individual event to keep it consistent with the rest of the collectors.  The base_manager event_catcher expects that each item in the queue is a single event.  This might work but best to not have too many surprises.